test/test_testing.py

```diff
@@ -146,43 +137,45 @@ def reset_can_connect():
             ''')
         self.addCleanup(harness.cleanup)
 
-        self.assertRaises(RuntimeError, harness.set_can_connect, 'foo', False)
-
         harness.begin()
         c = harness.model.unit.get_container('foo')
+
+        self.assertFalse(c.can_connect())
+        with self.assertRaises(pebble.ConnectionError):
+            c.get_plan()
+
+        harness.set_can_connect('foo', True)
         self.assertTrue(c.can_connect())
 
-    def test_simulate_can_connect(self):
-        tmp = ops.testing.SIMULATE_CAN_CONNECT
-        ops.testing.SIMULATE_CAN_CONNECT = True
+        harness.set_can_connect('foo', False)
+        self.assertFalse(c.can_connect())
 
-        def reset_can_connect():
-            ops.testing.SIMULATE_CAN_CONNECT = tmp
-        self.addCleanup(reset_can_connect)
+        harness.container_pebble_ready('foo')
+        self.assertTrue(c.can_connect())
+        c.get_plan()  # shouldn't raise ConnectionError
 
+    def test_can_connect_begin_with_initial_hooks(self):
```

Comment:
I think this test should be updated not to test just that you 'can connect' but that begin_with_initial_hooks actually triggers the pebble-ready events for defined containers.

Reply:
Yes definitely. Thanks.